modules/vebal/voting-gauges.repository.ts

```diff
@@ -42,32 +37,168 @@ type SubGraphGauge = {
+            })
+            .then(Number);
+
+        const contracts = Array.from({ length: totalGauges }, (_, index) => ({
+            abi: gaugeControllerAbi as any,
+            address: gaugeControllerAddress as `0x${string}`,
+            functionName: 'gauges',
+            args: [index],
+        }));
+
+        const addresses = await this.viemClient
+            .multicall({ contracts })
+            .then((results) =>
+                results.map((result) =>
+                    result.status === 'success' ? (result.result as string).toLowerCase() : undefined,
+                ),
+            );
+
+        return addresses as string[];
+    }
+
+    async fetchTypeNames() {
+        const totalGaugesTypes = await this.viemClient
+            .readContract({
+                address: gaugeControllerAddress as `0x${string}`,
+                functionName: 'n_gauge_types',
+                abi: gaugeControllerAbi,
+            })
+            .then(Number);
+
+        const contracts = Array.from({ length: totalGaugesTypes }, (_, index) => ({
+            abi: gaugeControllerAbi as any,
+            address: gaugeControllerAddress as `0x${string}`,
+            functionName: 'gauge_type_names',
+            args: [index],
+        }));
+
+        const typeNames = await this.viemClient.multicall({ contracts }).then((results) =>
+            results.map((result, index) => {
+                const name = result.status === 'success' ? result.result : undefined;
+                if (!name) {
+                    console.error(`Failed to fetch gauge type name at index ${index}`);
+                }
+                return name as string;
+            }),
+        );
 
-    async getVotingGaugeAddresses(): Promise<string[]> {
-        const totalGauges = Number(formatFixed(await this.getGaugeControllerContract().n_gauges()));
-        return await this.fetchGaugeAddresses(totalGauges);
+        return typeNames;
     }
 
-    async fetchOnchainVotingGauges(gaugeAddresses: string[]): Promise<VotingGauge[]> {
-        const totalGaugesTypes = Number(formatFixed(await this.getGaugeControllerContract().n_gauge_types()));
+    /**
+     * We need to use multicall3 with allowFailures=true because many of the root contracts do not have getRelativeWeightCap function defined
+     */
+    async fetchRelativeWeightCaps(gaugeAddresses: string[]): Promise<Record<string, string | undefined>> {
+        const contracts = gaugeAddresses.map((address) => ({
+            abi: rootGaugeAbi as any,
+            address: address as `0x${string}`,
+            functionName: 'getRelativeWeightCap',
+        }));
+
+        const results = await this.viemClient.multicall({ contracts, allowFailure: true });
+        const caps = gaugeAddresses.map((address, index) => {
+            const result = results[index];
+            const cap = result.status === 'success' ? result.result : undefined;
+            if (!cap) {
+                // Too many know errors to log
```

Comment:
I meant to explain what was already mentioned in the the function doc. `getRelativeWeightCap` isn't widely supported, so most of the calls with fail and we don't need to log that, because it's expected. I'll just remove the comment.